src/System.Runtime.Serialization.Xml/tests/Performance/PerformanceTestsCommon.cs

```diff
@@ -0,0 +1,210 @@
+            yield return new object[] { 1000, TestType.ListOfInt, 1024 };
+            yield return new object[] { 1, TestType.ListOfInt, 1024 * 1024 };
+            yield return new object[] { 1000, TestType.Dictionary, 128 };
+            yield return new object[] { 100, TestType.Dictionary, 1024 };
+            yield return new object[] { 10, TestType.SimpleType, 1 };
+            yield return new object[] { 1, TestType.SimpleType, 15 };
+            yield return new object[] { 10000, TestType.ISerializable, -1 };
+            yield return new object[] { 10000, TestType.XmlElement, -1 };
+        }
+
+        #endregion
+
+        #region Methods to create object for serialization tests
+
+        public static SimpleTypeWihtMoreProperties CreateSimpleTypeWihtMoreProperties(int height, int parentId, int currentId, int collectionSize, int childListSize)
+        {
+            int index = parentId * childListSize + (currentId + 1);
+            var obj = new SimpleTypeWihtMoreProperties()
+            {
+                IntProperty = index,
+                StringProperty = index + " string value",
+                EnumProperty = (MyEnum)(index % (Enum.GetNames(typeof(MyEnum)).Length)),
+                CollectionProperty = new List<string>(collectionSize),
+                SimpleTypeList = new List<SimpleTypeWihtMoreProperties>(childListSize)
+            };
+            for (int i = 0; i < collectionSize; ++i)
+            {
+                obj.CollectionProperty.Add(index + "." + i);
+            }
+            if (height > 1)
+            {
+                for (int i = 0; i < childListSize; ++i)
+                {
+                    obj.SimpleTypeList.Add(CreateSimpleTypeWihtMoreProperties(height - 1, index, i, collectionSize, childListSize));
+                }
+            }
+            return obj;
+        }
+
+        public static byte[] CreateByteArray(int size)
+        {
+            byte[] obj = new byte[size];
+            for (int i = 0; i < obj.Length; ++i)
+            {
+                unchecked
+                {
+                    obj[i] = (byte)i;
+                }
+            }
+            return obj;
+        }
+
+        public static Dictionary<int, string> CreateDictionaryOfIntString(int count)
+        {
+            Dictionary<int, string> dictOfIntString = new Dictionary<int, string>(count);
+            for (int i = 0; i < count; ++i)
+            {
+                dictOfIntString[i] = i.ToString();
+            }
+
+            return dictOfIntString;
+        }
+
+        public static List<int> CreateListOfInt(int count)
+        {
+            return Enumerable.Range(0, count).ToList();
+        }
+
+        public static object GetSerializationObject(TestType testType, int testSize)
+        {
+            Object obj = null;
+            switch (testType)
+            {
+                case TestType.ByteArray:
+                    obj = CreateByteArray(testSize);
```

Comment:
When https://github.com/Microsoft/xunit-performance/issues/124 is resolved we should eliminate TestType enum.